src/main/java/com/fasterxml/jackson/databind/deser/std/UntypedObjectDeserializer.java

```diff
@@ -191,18 +218,32 @@ public JsonDeserializer<?> createContextual(DeserializationContext ctxt,
         // 14-Jun-2017, tatu: [databind#1625]: may want to block merging, for root value
         boolean preventMerge = (property == null)
                 && Boolean.FALSE.equals(ctxt.getConfig().getDefaultMergeable(Object.class));
+        // 31-Aug-2024: [databind#4680] Allow custom key deserializer for Object.class deserialization
+        KeyDeserializer keyDeser = ctxt.findKeyDeserializer(SimpleType.constructUnsafe(Object.class), property);
```

Comment:
I think this needs to see if we get the "default" key deserializer -- I forget how exactly it is done, maybe regular `MapDeserializer` has an example -- and if so, ignore it (re-set to `null`).

Reply:
Thanks! Added [commit](https://github.com/FasterXML/jackson-databind/pull/4684/commits/a1fe9cc888a4bffb64a604b500a7cc7577b02173) according to your suggestio and we are down to one last case 👌🏼👌🏼

```log
[ERROR] Errors: 
[ERROR]   UntypedDeserializationTest.testUntypedWithCustomScalarDesers:263 » NullPointer
```

Reply:
Ah, simple mistake, didn't check if we had _customKeyDeserializer not null